src/controllers/dream-controller.ts

```diff
@@ -0,0 +1,340 @@
+     * @async
+     */
+    public async getCommentDreamHandler(req:Request, res:Response): Promise<Response>{
+        try{
+            const dream = await this.db.dreams.findById(req.params.id);
+            if (dream == null) {
+                return res.status(404).send(this.container.errors.formatErrors({
+                    error: 'not_found',
+                    error_description: 'Dream not found'
+                }));
+            }   return res.status(200).send({comments: dream.comments})
+        }catch(err){
+            return res.status(500).send(this.container.errors.formatServerError());
+        }
+    }
+     /**
+     * Get a specific comment in a specific dream.
+     * 
+     * Path : `GET /dreams/:id/comments/:id`
+     * 
+     * @param req Express request
+     * @param res Express response
+     * @async
+     */
+    public async getSpecificCommentDreamHandler(req:Request, res:Response): Promise<Response>{
+        try{
+            const dream = await this.db.dreams.findById(req.params.id);
+            if (dream == null) {
+                return res.status(404).send(this.container.errors.formatErrors({
+                    error: 'not_found',
+                    error_description: 'Dream not found'
+                }));
+            } 
+            const comment = dream.comments.find(comment => comment.id === req.params.commentId)
+            if(comment == null){
+                return res.status(404).send(this.container.errors.formatErrors({
+                    error: 'not_found',
+                    error_description: 'Comment not found'
+                }));
+            } 
+                return res.status(200).send({comment})
+        }catch(err){
+            return res.status(500).send(this.container.errors.formatServerError());
+        }
+    }
+     /**
+     * Creates a new comment.
+     * 
+     * Path : `POST /dreams/:id/comments`
+     * 
+     * @param req Express request
+     * @param res Express response
+     * @async
+     */
+    public async createCommentDreamHandler(req: Request, res: Response): Promise<Response> {
+        
+        try {
+            if (!await this.db.dreams.exists({_id:req.params.id})){
+                return res.status(404).send(this.container.errors.formatErrors({
+                    error: 'not_found',
+                    error_description: 'Dream not found'
+                }));
+            };
+            const comment = await this.db.comments.create({
+                content:   req.body.content,
+                author:    req.body.author,
+                parent:    req.body.comments,
+                dream:     req.body.id
+            });
+            return res.status(201).send({
+                id: comment.id,
+                links: [{
+                    rel: 'Gets the created comment',
+                    action: 'GET',
+                    href: `${req.protocol}://${req.get('host')}${this.rootUri}/${comment.id}`
```

Comment:
Route mal formattée, il manque le dream ID :
`${req.protocol}://${req.get('host')}${this.rootUri}/${req.body.id}/${comment.id}`